source/network/ODServer.cpp

```diff
@@ -613,6 +613,8 @@ bool ODServer::processClientNotifications(ODSocketClient* clientSocket)
 
             Seat* seat = seats[0];
             seat->setPlayer(curPlayer);
+            //This makes sure the player is deleted on exit.
+            gameMap->addPlayer(curPlayer);
```

Comment:
Yeah the player was not deleted in the editor mode (according to valgrind). As you are saying, I don't know if it was the best solution, the code was a bit hard to follow. Feel free to suggest something else. In gamemode the players are added to the  gamemap later, and deleted when it's cleared but it didn't look like that was happening in editormode.